src/v/archival/service.cc

```diff
@@ -169,15 +169,10 @@ ss::future<> scheduler_service_impl::start() {
 ss::future<> scheduler_service_impl::stop() {
     vlog(_rtclog.info, "Scheduler service stop");
     _timer.cancel();
-    std::vector<ss::future<>> outstanding;
+    co_await _gate.close();
     for (auto& it : _archivers) {
-        outstanding.emplace_back(it.second->stop());
+        co_await it.second->stop();
     }
```

Comment:
can we use `when_all_succeed` here to wait on stopping the archivers in parallel?

Reply:
will add it in a next PR